### PR TITLE
CSUB-1073: Prepare for release v3.21.0 / Public Testnet

### PR DESCRIPTION
**Pull request by atodorov**
# Description of proposed changes

Syncronize latest changes in `dev` with the `testnet` branch in a preparation for an upcoming release.

---

Practical tips for PR review & merge:

- [ ] All GitHub Actions report PASS
- [ ] Newly added code/functions have unit tests
  - [ ] Coverage tools report all newly added lines as covered
  - [ ] The positive scenario is exercised
  - [ ] Negative scenarios are exercised, e.g. assert on all possible errors
  - [ ] Assert on events triggered if applicable
  - [ ] Assert on changes made to storage if applicable
- [ ] Modified behavior/functions - try to make sure above test items are covered
- [ ] Integration tests are added if applicable/needed

### CODE REVIEW

**Review by atodorov — COMMENTED** [review]
The failure in [danger-will-brick-the-blockchain](https://github.com/gluwa/creditcoin3/actions/runs/8233764030/job/22513949093?pr=267) is a false positive and can be ignore. It matches on `MILLISECONDS_PER_BLOCK` which is part of the overall `git diff` because neighbouring lines have been changed but hasn't actually changed the value of this constant.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -294,27 +300,38 @@ jobs:
...
           command: build
-          args: --release
+          args: --release ${{ steps.compiler-flags.outputs.extra }}
 
       - name: Upload creditcoin-node binary
         uses: actions/upload-artifact@v4
         with:
-          name: creditcoin-node
+          name: creditcoin-node-${{ matrix.target }}
           path: target/release/creditcoin3-node
```
Comment (beqaabu) [inline .github/workflows/ci.yml:319]: Do we want to keep creditcoin3 name here or change it to just creditcoin? I see that its an upload job but still

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -294,27 +300,38 @@ jobs:
...
 
       - name: Upload WASM runtime
         uses: actions/upload-artifact@v4
         with:
-          name: creditcoin_node_runtime.compact.compressed.wasm
+          name: creditcoin_node_runtime.compact.compressed.wasm-${{ matrix.target }}
           path: target/release/wbuild/creditcoin3-runtime/creditcoin_next_runtime.compact.compressed.wasm
```
Comment (beqaabu) [inline .github/workflows/ci.yml:325]: same as above regarding the naming

**Review by pLabarta — APPROVED** [review]

**Review by beqaabu — APPROVED** [review]
LGTM just 2 small Qs

**Pull request merged**